.coveragerc

```diff
@@ -0,0 +1,5 @@
+[report]
+exclude_lines=
+    pragma: no cover
+    def __enter__
```

Comment:
Could you explain a bit why enter and exit should be excluded?

I think it should go the other way - if the goal is for Collector to be usable with the context manager protocol, there should be some tests verifying that this is the case.  enter/exit would be naturally covered by those tests and wouldn't need to be excluded.

There should probably be a test of this nature:

- register a backend with an \_\_enter\_\_ / \_\_exit\_\_
- do `with Collector.get():`
- verify that my \_\_enter\_\_ / \_\_exit\_\_ were called

It's non-trivial enough to warrant testing because it relies on the correct implementation of the CollectorProxy class.

Reply:
My goal was to avoid writing trivial tests just to exercise the no-op `__enter__` and `__exit__` methods that I added to the existing backends for the sake of coverage. Since I've dropped those methods from the existing backends based on our other discussion, I no longer feel it is necessary to exclude these lines from coverage so I dropped the .coveragerc file as well. I figure we can add this file back as needed in the future.